src/main/java/org/cyclops/integratedterminals/client/gui/container/ContainerScreenTerminalStorage.java

```diff
@@ -690,15 +699,19 @@ public boolean mouseReleased(double mouseX, double mouseY, int mouseButton) {
             this.clicked = false;
             Optional<ITerminalStorageTabClient<?>> tabOptional = getSelectedClientTab();
             if (tabOptional.isPresent()) {
+                ITerminalStorageTabClient tab = tabOptional.get();
                 int slot = getStorageSlotIndexAtPosition(mouseX, mouseY);
                 Slot playerSlot = getSlotUnderMouse();
 
-                // Handle clicks on storage slots
+                // Handle clicks on storage slots or container slots when storage has an active slot (item on cursor) or the crafting result slot
                 boolean hasClickedOutside = this.hasClickedOutside(mouseX, mouseY, this.leftPos, this.topPos, mouseButton);
                 boolean hasClickedInStorage = this.hasClickedInStorage(mouseX, mouseY);
-                if (tabOptional.get().handleClick(getMenu(), getMenu().getSelectedChannel(), slot, mouseButton,
-                        hasClickedOutside, hasClickedInStorage, playerSlot != null ? playerSlot.index : -1)) {
-                    return true;
+                boolean hasActiveSlot = tab.getActiveSlotId() >= 0;
+                if(hasClickedInStorage || hasActiveSlot || isSlotUnderMouseCraftingResult()) {
```

Comment:
Do we really need this if-condition around the `tab.handleClick`?
I'd like to keep this code as generic as possible, so that tabs can provide their own implementations of this.
I guess it should be possible to move this slot into tab-specific implementations?

Reply:
We'd need an extra argument to `#handleClick` like `isQuickMove` so the click isn't handled twice. We could use `isShifted` but that's not safe because the client could use other controls to trigger quick move.

Reply:
The newest commit added the `isQuickMove` parameter. That resolved the other change requests because the tab implementations can decide how to handle quickmoves in their `#handleClick`